acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/signing/PrivateTransactionAcceptanceTest.java

```diff
@@ -0,0 +1,131 @@
+import tech.pegasys.ethsigner.tests.signing.contract.generated.SimpleStorage;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URL;
+
+import com.github.dockerjava.api.DockerClient;
+import com.google.common.io.Resources;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PrivateTransactionAcceptanceTest {
+
+  private static final DockerClient DOCKER = new DockerClientFactory().create();
+
+  private static final String ENCLAVE_PUBLIC_KEY1 = "enclave_key1.pub";
+  private static final String ENCLAVE_PUBLIC_KEY2 = "enclave_key2.pub";
+
+  private Node ethNode;
+  private Signer ethSigner;
+
+  private Account richBenefactor() {
+    return ethSigner.accounts().richBenefactor();
+  }
+
+  private String enclavePublicKey1() {
+    return readResource(ENCLAVE_PUBLIC_KEY1);
+  }
+
+  private String enclavePublicKey2() {
+    return readResource(ENCLAVE_PUBLIC_KEY2);
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  private String readResource(final String resourceName) {
+    final URL resource = Resources.getResource(resourceName);
+    try {
+      return new String(Resources.toString(resource, UTF_8).getBytes(UTF_8), UTF_8);
+    } catch (final IOException e) {
+      throw new RuntimeException("Unable to load resource " + resourceName);
+    }
+  }
+
+  @Before
+  public void setUp() {
+    Runtime.getRuntime().addShutdownHook(new Thread((this::tearDown)));
+
+    final NodeConfiguration nodeConfig =
+        new NodeConfigurationBuilder()
+            .withPrivacyEnabled()
+            .withPrivacyPublicKey(ENCLAVE_PUBLIC_KEY2)
+            .build();
+    ethNode = new PantheonNode(DOCKER, nodeConfig);
+    ethNode.start();
+    ethNode.awaitStartupCompletion();
+
+    final SignerConfiguration signerConfig = new SignerConfigurationBuilder().build();
+    ethSigner = new Signer(signerConfig, nodeConfig, ethNode.ports());
+    ethSigner.start();
+    ethSigner.awaitStartupCompletion();
+  }
+
+  @After
+  public void tearDown() {
+    if (ethNode != null) {
+      ethNode.shutdown();
+    }
+
+    if (ethSigner != null) {
+      ethSigner.shutdown();
+    }
+  }
+
+  @Test
+  public void deployContract() {
+    final PrivateTransaction contract =
+        PrivateTransaction.createContractTransaction(
+            richBenefactor().address(),
+            richBenefactor().nextNonceAndIncrement(),
+            GAS_PRICE,
+            GAS_LIMIT,
+            BigInteger.ZERO,
+            SimpleStorage.BINARY,
+            enclavePublicKey1(),
+            singletonList(enclavePublicKey2()),
+            RESTRICTED);
+
+    final SignerResponse<JsonRpcErrorResponse> signerResponse =
+        ethSigner.contracts().submitExceptionalPrivateTransaction(contract);
+    // We expect this to fail with enclave error as we don't have orion running. If rlp decode fails
+    // then we would get a different error
+    assertThat(signerResponse.status()).isEqualTo(BAD_REQUEST);
```

Comment:
probably don't care if its BAD_REQUEST - i.e. so long as it failed.

Reply:
done